### ISSUE TITLE
One-to-one relation on same ID does not create child record

### ISSUE THREAD

**Issue by amoe** [friemen/aggregate#5]
Hi, I want to use aggregate to save an inheritance-style relation, where the child table shares a key with the parent table.  I give an example script below, using postgresql.  The schema is obviously not the real thing, in reality this is a list of user actions that are differentiated by type but need to be handled uniformly in some places.

```
(ns aggregate-one-to-one.core
  (:gen-class)
  (:require [aggregate.core :as agg]
    [clojure.java.jdbc :as jdbc]))


(def db-host "localhost")
(def db-port 5432)
(def db-name "test")
(def db-spec {:classname "org.postgresql.Driver"
      :subprotocol "postgresql"
      :subname (str "//" db-host ":" db-port "/" db-name)
      :user "test"
      :password "test"})

(def model
  (agg/make-er-config
   (agg/entity :animal)
   (agg/entity :cow
       (agg/->1 :animal :animal {:fk-kw :id}))))

(def ddl ["CREATE TABLE animal (id SERIAL PRIMARY KEY, phylum VARCHAR(80));"
      "CREATE TABLE cow (id INTEGER NOT NULL, breed VARCHAR(80), FOREIGN KEY (id) REFERENCES animal (id));"])


(defn -main
  "I don't do a whole lot ... yet."
  [& args]
  (doseq [t ddl]
(jdbc/execute! db-spec [t]))

  (agg/save! model db-spec :cow {:animal {:phylum "Chordata"} :breed "Friesian"})
  (println "Hello, World!"))
```

When you run this code, a parent record is correctly inserted in the `animal` table.  But no record is inserted in the `cow` table.  I have an idea that this might be due to the reuse of the name `id` in the schema for the primary key, or maybe my entity model is defined incorrectly.  Is there any fix or workaround for this?

**Comment by friemen** [comment]
I think the bug is that the check whether a record has already been persisted is conducted AFTER foreign-keys of :one> relationsships have been assoc'd. In your case (foreign-key is the same as primary key), aggregate wrongly assumes that the :cow record needs an update instead of an insert.

I'll reproduce your issue with a unit test and will try to fix it.

**Comment by friemen** [comment]
Fixed, released version 1.1.0 https://clojars.org/aggregate/versions/1.1.0
